src/Akeneo/Connectivity/Connection/back/tests/Unit/spec/Infrastructure/EventSubscriber/ReadProductsEventSubscriberSpec.php

```diff
@@ -62,29 +73,84 @@ public function it_saves_read_products_events(
         )->shouldBeCalled();
     }
 
-    public function it_does_not_save_read_products_events_for_not_collectable_connection(
+    public function it_saves_read_products_events_with_an_events_api_event(
         $connectionContext,
-        $updateDataDestinationProductEventCountHandler
+        $updateDataDestinationProductEventCountHandler,
+        Connection $connection,
+        ConnectionRepository $connectionRepository
     ): void {
-        $connectionContext->isCollectable()->willReturn(false);
+        $readProductsEvent = new ReadProductsEvent(3, ReadProductsEvent::EVENTS_API_TYPE, 'ecommerce');
+        $connection->hasDataDestinationFlowType()->willReturn(true)->shouldBeCalledTimes(1);
```

Comment:
```suggestion
        $connection->hasDataDestinationFlowType()->willReturn(true);
```
Same here. No matters if it's called or not, it will not change the behavior the class. Please remove all the should and shouldNot. The real test is this one https://github.com/akeneo/pim-community-dev/pull/14503/files#diff-88334017883f497d59648ee8218b74001a064d2e5224d0337d3d0b29b5432d51R94-R100